cmd/backup-manager/app/backup/manager.go

```diff
@@ -337,3 +341,169 @@ func (bm *Manager) performBackup(ctx context.Context, backup *v1alpha1.Backup, d
 		Status: corev1.ConditionTrue,
 	}, updateStatus)
 }
+
+// performLogBackup execute log backup commands according to backup cr.
+func (bm *Manager) performLogBackup(ctx context.Context, backup *v1alpha1.Backup) error {
+	var (
+		err          error
+		reason       string
+		resultStatus *controller.BackupUpdateStatus
+	)
+
+	if err := bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
+		Command: v1alpha1.LogSubCommandType(bm.SubCommand),
+		Type:    v1alpha1.BackupPrepare,
+		Status:  corev1.ConditionTrue,
+	}, nil); err != nil {
+		return err
+	}
+
+	// start/stop/truncate log backup
+	switch bm.SubCommand {
+	case string(v1alpha1.LogStartCommand):
+		resultStatus, reason, err = bm.startLogBackup(ctx, backup)
+	case string(v1alpha1.LogStopCommand):
+		resultStatus, reason, err = bm.stopLogBackup(ctx, backup)
+	case string(v1alpha1.LogTruncateCommand):
+		resultStatus, reason, err = bm.truncateLogBackup(ctx, backup)
+	default:
+		return fmt.Errorf("log backup %s unknown log subcommand %s", bm, bm.SubCommand)
+	}
+
+	// handle error
+	if err != nil {
+		errs := make([]error, 0)
+		errs = append(errs, err)
+		// update failed status
+		uerr := bm.StatusUpdater.Update(backup, &v1alpha1.BackupCondition{
```

Comment:
Better add retry to status update since there may be racing condition for updating BackupCR. If don't have retry inside backupmanager, we may need to retry from operator. Am I correct? If I am right, please add status update everywhere.

Reply:
yes, you are right. I also found this problem when coding, and then read code in [here](https://github.com/pingcap/tidb-operator/pull/4682/files#diff-87b9f36ed20d95aff7bf42e5692e5f0fad07d7244b6eb097d82fc6923fe3096eR97-R109), it can retry and get latest cr, so it can work if we don't depend on original cr(especially cr spec) to update cr info.